libnetdata/config/appconfig.c

```diff
@@ -472,6 +477,23 @@ int appconfig_load(struct config *root, char *filename, int overwrite_used)
             co = appconfig_section_find(root, s);
             if(!co) co = appconfig_section_create(root, s);
 
+            if(co && section_name && overwrite_used && section_hash == co->hash && !strcmp(section_name, co->name)) {
+                config_section_wrlock(co);
+                struct config_option *cv2 = co->values;
+                while (cv2) {
+                    struct config_option *save = cv2->next;
+                    struct config_option *found = appconfig_option_index_del(co, cv2);
+                    if(found != cv2)
+                        error("INTERNAL ERROR: Cannot remove '%s' from  section '%s', it was not inserted before.",
+                               cv2->name, co->name);
+
+                    free(cv2);
+                    cv2 = save;
+                }
+                co->values = NULL;
+                config_section_unlock(co);
+            }
+
```

Comment:
Github UI won't let me attach comment to L505-L516:

This just needs one part from the code you wrote in the separate function: if we are reloading a single section, skip overwriting any fields outside that section. i.e.

```
[general]
    ... ignore any changes don't write anything....
[host labels]
   ... write all the stuff in here...
[plugins]
   ... ignore everything, don't update any changes...
```

Reply:
You are right, I forgot to bring this from the function I removed, I brought it now. Thank you!